subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationStep.java

```diff
@@ -99,20 +99,22 @@ public int stepsCount() {
         isolateTransformerParameters(fingerprinterRegistry);
 
         Try<ArtifactTransformDependencies> resolvedDependencies = dependenciesResolver.forTransformer(transformer);
-        return resolvedDependencies.getSuccessfulOrElse(dependencies -> {
-            ImmutableList<File> inputArtifacts = subjectToTransform.getFiles();
-            if (inputArtifacts.isEmpty()) {
-                return CacheableInvocation.cached(Try.successful(subjectToTransform.createSubjectFromResult(ImmutableList.of())));
-            } else if (inputArtifacts.size() > 1) {
-                return CacheableInvocation.nonCached(() ->
+        return resolvedDependencies
+            .tryMap(dependencies -> {
```

Comment:
Why do we want to catch exceptions here? When a failure happens here then it seems to be in the Gradle code, so I think the whole build should fail, not only the transform.

Reply:
Ah, I merged too quick.

It's a good point. Here's the resolution: https://github.com/gradle/gradle/pull/10148